File: client/app/app/src/main/java/com/mascot/app/data/model/Quest.kt
package com.mascot.app.data.model

data class Quest(
    val id: String,
    val title: String,
    val description: String,
    val location: String? = null,
    val duration: String? = null,
    val type: String? = null,
    val imageUrl: String? = null,
    val isCompleted: Boolean = false
)

File: server/models/Quest.js
import mongoose from "mongoose";

const QuestSchema = new mongoose.Schema({
  region: { type: String, required: true },  
  id: { type: String, required: true },
  title: { type: String, required: true },
  description: { type: String, required: true },
  location: { type: String, required: true },
  category: { type: String, required: true },
  coordinates: {
    lat: Number,
    lng: Number
  }
});

export default mongoose.model("Quest", QuestSchema);
